upup/pkg/fi/cloudup/vsphere/vsphere_cloud.go

```diff
@@ -81,17 +88,34 @@ func NewVSphereCloud(spec *kops.ClusterSpec) (*VSphereCloud, error) {
 	}
 	// Add retry functionality
 	c.RoundTripper = vim25.Retry(c.RoundTripper, vim25.TemporaryNetworkError(5))
-	vsphereCloud := &VSphereCloud{Server: server, Datacenter: datacenter, Cluster: cluster, Username: username, Password: password, Client: c}
+	vsphereCloud := &VSphereCloud{Server: server, Datacenter: datacenter, Cluster: cluster, Username: username, Password: password, Client: c, CoreDNSServer: dns_server, DNSZone: dns_zone}
 	glog.V(2).Infof("Created vSphere Cloud successfully: %+v", vsphereCloud)
 	return vsphereCloud, nil
 }
 
 func (c *VSphereCloud) DNS() (dnsprovider.Interface, error) {
-	glog.Warning("DNS() not implemented on VSphere")
-	provider, err := dnsprovider.GetDnsProvider(k8sroute53.ProviderName, nil)
-	if err != nil {
-		return nil, fmt.Errorf("Error building (k8s) DNS provider: %v", err)
+	// TODO: this is a temporary flag to toggle between CoreDNS and Route53, before CoreDNS is stable
+	dns_provider := os.Getenv("VSPHERE_DNS")
+
+	var provider dnsprovider.Interface
+	var err error
+	if dns_provider == "coredns" {
```

Comment:
I can claim this on the beginning of this file as a constant. But this one is just for temporary use. We will remove it soon once CoreDNS is stable.